London| Elhadj Abdoul Diallo|Module-Data-Flows| WEEK3 - cowsay_&_weather-app

eediallo opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist
I have done the ```consway``` and ```weather-app``` challenges. The weather app might not work as intended because I chose not to make the  ```weatherAPIKey``` and ``` unsplashAccessKey``` keys public. Please request access to the keys [here](https://docs.google.com/document/d/1kB_V1q4qH3Ov4bA3an_0sV6adH4ulLOynQm3HAbgFNA/edit?usp=sharing)

eediallo commented:
Please note that the weather app might not work as intended because I chose not to make the  ```weatherAPIKey``` and ``` unsplashAccessKey``` keys public. Please request access to the keys [here](https://docs.google.com/document/d/1kB_V1q4qH3Ov4bA3an_0sV6adH4ulLOynQm3HAbgFNA/edit?usp=sharing)

eediallo commented:
Hi @tyzia ! Could you please let me know when you will be able to complete this PR review?

mjpeet reviewed (commented):
excellent work, you go above and beyond with solving these challenges!

with regards to api keys - the volunteers need to look at working code, and so when they checkout your branch/changes, the code needs to run as is. (in a normal work environment usually you also create a step by step document, how to apply your key, so people not familiar with your project can test it)

eediallo commented on challenges/challenge-weather-app/script.js:
```diff
@@ -0,0 +1,218 @@
+  static weatherAPIKey = config.weather_API_Key;
+
+  async fetchWeatherData() {
+    const url = `http://api.openweathermap.org/data/2.5/weather?q=${this.city}&appid=${Weather.weatherAPIKey}`;
```
Yes I am going to add proper error handling.

eediallo commented on challenges/challenge-weather-app/script.js:
```diff
@@ -0,0 +1,218 @@
+const emptyOrNumericCity = new EmptyOrNumericCity();
+
+//==========Weather Class======================
+class Weather {
```
I could move each class in a separate file to make it easy to read

eediallo commented:
@mjpeet ! Thank you for your constructive feedback. I have refactored the code by moving each class into its own file and implemented better handling for fetching both the weather data and the photos. This improves the code's readability and enhances security.
